drivers/SmartThings/zigbee-window-treatment/src/mic/init.lua

```diff
@@ -0,0 +1,303 @@
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local log = require "log"
+local capabilities = require "st.capabilities"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local window_shade_defaults = require "st.zigbee.defaults.windowShade_defaults"
+local WindowCovering = zcl_clusters.WindowCovering
+local cluster_base = require "st.zigbee.cluster_base"
+local utils = require "st.utils"
+local data_types = require "st.zigbee.data_types"
+local window_preset_defaults = require "st.zigbee.defaults.windowShadePreset_defaults"
+
+
+
+
+local GLYDEA_MOVE_THRESHOLD = 3
+
+local MOVE_LESS_THAN_THRESHOLD = "_sameLevelEvent"
+
+
+
+local micronic_window_shade_FINGERPRINTS = {
+  { mfr = "micronic-ko", model = "acm301" }
+  
+
+}
+
+local is_micronic_window_shade = function(opts, driver, device)
+  for _, fingerprint in ipairs(micronic_window_shade_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+
+end
+
+
+
+
+
+local INVERT_CLUSTER = 0xFC00
+local INVERT_CLUSTER_ATTRIBUTE = 0x0000
+ 
+
+
+
+local function invert_preference_handler(device)
+
+
+
+
+  local window_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+    capabilities.windowShadeLevel.shadeLevel.NAME) or 0
+  local window_shade = capabilities.windowShade.windowShade
+  device:emit_event(window_level == 100 and window_shade.closed() or window_shade.open())
+  device:emit_event(capabilities.windowShadeLevel.shadeLevel(100 - window_level))
+
+
+  log.debug(string.format(" invert_preference_handler~~  (%s) (%d)", device, window_level))
+
+
+ 
+  local invert_value = device.preferences.invert
+  local invert_cluster_cmd = cluster_base.write_manufacturer_specific_attribute(device,
+    INVERT_CLUSTER,
+    INVERT_CLUSTER_ATTRIBUTE,
+    0x0000, data_types.Boolean,
+    invert_value)
+  device:send(invert_cluster_cmd)
+end
+
+local function info_changed(driver, device, event, args)
+
+  log.debug(string.format(" info_changed~~  (%s) ", device))
+
+
+
+  if device.preferences ~= nil and device.preferences.invert ~= args.old_st_store.preferences.invert then
+    invert_preference_handler(device)
+  end
+end
+
+local set_window_shade_level = function(level)
+
+  log.debug(string.format(" set_window_shade_level  (%s) ", level))
+
+
+  return function(driver, device, cmd)
+    log.debug(string.format(" set_window_shade_level (%s) (%s) (%s) ", driver, device, level))
+    device:send_to_component(cmd.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+  end
+end
+
+
+
+
+
+local WindowCovering = zcl_clusters.WindowCovering
+
+local SHADE_SET_STATUS = "shade_set_status"
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+
+
+
+
+
+ 
```

Comment:
there's a lot of extra whitespace throughout this file. Could you please clean it up a bit, for readability's sake?